R/newstart_allowance.R

```diff
@@ -0,0 +1,169 @@
+#' @param per Specifies the timeframe in which payments will be made. Can either take value "fortnight" or "annual".
+#' @source \url{http://classic.austlii.edu.au/au/legis/cth/consol_act/ssa1991186/s1068.html}
+#' @export newstart_allowance
+
+#historical rates single: http://guides.dss.gov.au/guide-social-security-law/5/2/1/20
+  #married: http://guides.dss.gov.au/guide-social-security-law/5/2/1/30
+#better copy of rates and reductions: https://www.humanservices.gov.au/sites/default/files/co029-1603en.pdf
+newstart_allowance <- function(fortnightly_income = 0,
+                               annual_income = 0,
+                               has_partner = FALSE,
+                               partner_pensioner = FALSE,
+                               n_dependants = 0,
+                               nine_months = FALSE,
+                               isjspceoalfofcoahodeoc = FALSE,
+                               principal_carer = FALSE,
+                               fortnightly_partner_income = 0,
+                               annual_partner_income = 0,
+                               age = 22,
+                               fy.year = "2015-16",
+                               assets_value = 0,
+                               homeowner = FALSE,
+                               lower = 102,
+                               upper = 252,
+                               taper_lower = 0.5,
+                               taper_upper = 0.6,
+                               taper_principal_carer = 0.4,
+                               per = "fortnight") {
+  
+  if (fy.year != "2015-16") {
+    stop('`fy.year` can only take value "2015-16" for now')
+  }
+  
+  if(any(!(per %in% c("fortnight", "annual")))){
+    stop('`per`` can only take values "fortnight" or "annual"')
+  }
+  
+  if(dim(table(lengths(mget(ls())))) >2 ){
+    stop('inputs are not of the same length')
+  }
+  
+  input <- data.table(do.call(cbind.data.frame, mget(ls())))
+  
+  if(any(input[, if_else(fortnightly_income > 0 & annual_income > 0, TRUE, FALSE)])){
+    stop('cannot have inputs for both `fortnightly_income` and `annual_income` for the same individual')
+  }
+  
+  if(any(input[, if_else(fortnightly_partner_income > 0 & annual_partner_income > 0, TRUE, FALSE)])){
+    stop('cannot have inputs for both `fortnightly_partner_income` and `annual_partner_income` for the same individual')
+  }
+  
+  if(any(input[, if_else(!has_partner & partner_pensioner, TRUE, FALSE)])){
+    stop('check conflciting values for `has_partner`` and `partner_pensioner`')
+  }
+  
+  #replace missing fortnightly income with annual income / 26
+  input[,'fortnightly_income'] <-
+    input[, if_else(annual_income > 0,
+                    annual_income / 26,
+                    fortnightly_income)]
+  input[,'fortnightly_partner_income'] <-
+    input[, if_else(annual_partner_income > 0,
+                    annual_partner_income / 26,
+                    fortnightly_income)]
```

Comment:
In `data.table` parlance, use `:=` to add a new column (or modify an existing one):

`input[, fortnightly_income := if_else(annual_income > 0, annual_income / 26, fortnightly_income)]`.